positron/components/ModuleLoader.jsm

```diff
@@ -42,28 +42,32 @@ const globalPaths = [
   'node/',
 ];
 
-/**
- * Mapping from module IDs (resource: URLs) to module objects.
- *
- * @keys {string} The ID (resource: URL) for the module.
- * @values {object} An object representing the module.
- */
-let modules = new Map();
+const windowLoaders = new WeakMap();
 
 /**
  * Construct a module importer (`require()` global function).
  *
  * @param requirer {Module} the module that will use the importer.
  * @return {Function} a module importer.
  */
-function Require(requirer) {
+
+function ModuleLoader(processType) {
+  /**
+   * Mapping from module IDs (resource: URLs) to module objects.
+   *
+   * @keys {string} The ID (resource: URL) for the module.
+   * @values {object} An object representing the module.
+   */
+  let modules = new Map();
+
   /**
    * Import a module.
    *
-   * @param path {string} the path to the module.
-   * @return {*} an `exports` object.
+   * @param  requirer {Object} the module importing this module.
+   * @param  path     {string} the path to the module being imported.
+   * @return          {Object} an `exports` object.
    */
-  return function require(path) {
+  let require = this.require = function(requirer, path) {
```

Comment:
to be clear, I was concerned that both variables/props were being declared. This could be confusing if one is used in one place, then another is used in another place. Thanks for removing the local one.